python-tr-ortools-region-allocation/main.py

```diff
@@ -0,0 +1,170 @@
+    objective.SetMinimization()
+
+    # Solves the problem.
+    status = solver.Solve()
+
+    # Get the assigned regions.
+    assignments_result = {
+        region["id"]: next(
+            hub["id"] for hub in input["hubs"] if assignments[(region["id"], hub["id"])].solution_value() > 0.5
+        )
+        for region in input["regions"]
+    }
+    solution = {
+        "assignments": assignments_result,
+        "hubs": input["hubs"],
+        "regions": input["regions"],
+    }
+
+    # Collect some statistics.
+    statistics = {
+        "run": {
+            "duration": time.time() - start_time,
+        },
+        "result": {
+            "duration": solver.WallTime() / 1000,
+            "value": solver.Objective().Value(),
+            "custom": {
+                "status": STATUS.get(status, "unknown"),
+                "variables": solver.NumVariables(),
+                "constraints": solver.NumConstraints(),
+            },
+        },
+    }
+
+    return solution, statistics
+
+
+def calculate_distance(region: dict, hub: dict) -> float:
+    """Calculates the distance between a region and a hub using the haversine formula."""
+    region_coords = region["center"]
+    hub_coords = hub["center"]
+    lat1, lon1 = region_coords["y"], region_coords["x"]
+    lat2, lon2 = hub_coords["y"], hub_coords["x"]
+
+    # Convert latitude and longitude from degrees to radians
+    lat1, lon1, lat2, lon2 = map(math.radians, [lat1, lon1, lat2, lon2])
+
+    # Haversine formula
+    dlat = lat2 - lat1
+    dlon = lon2 - lon1
+    a = math.sin(dlat / 2) ** 2 + math.cos(lat1) * math.cos(lat2) * math.sin(dlon / 2) ** 2
+    c = 2 * math.asin(math.sqrt(a))
+
+    # Radius of Earth in kilometers. Use 3956 for miles
+    r = 6371
+
+    # Calculate the result
+    return c * r
+
+
+if __name__ == "__main__":
+    # Parse the command line arguments.
+    parser = argparse.ArgumentParser(description="Region allocation problem solver.")
+    parser.add_argument("-input", type=str, help="Path to the input JSON file.")
+    parser.add_argument("-output", type=str, help="Path to the input JSON file.")
+    parser.add_argument("-provider", type=str, default="SCIP", help="Solver provider.")
+    parser.add_argument("-duration", type=int, default=60, help="Max duration in seconds.")
+    args = parser.parse_args()
+
+    # Read the input data.
+    with open(args.input) as f:
+        input_data = json.load(f)
+
+    # Solve the problem.
+    before = time.time()
+    output_data, statistics = solve(
+        input=input_data,
+        provider="SCIP",
+        duration=60,
+    )
+
+    # Write the output data.
+    with open(args.output, "w") as f:
+        json.dump(output_data, f, indent=2)
+
+    # >>> POST TRACKED RUN TO NEXTMV PLATFORM
+    # Note: Above code is intentionally not using nextmv features to better demonstrate
+    #       that tracking a run can be added easily to any existing code. You only need
+    #       lines below.
```

Comment:
I get it, but I would vote for using nextmv features anyhow 🤷🏻 .But I get the intention.

Reply:
The motivation is to show the tracked runs in context of a non-Nextmv-app, so, I would like to keep it this way. Particularly, for not diverging too much from the tech-talk for which this all is intended to serve as complimentary material.
However, I totally get that we are using the community-apps repo for sharing here, which may have different intentions. So, if we don't feel comfortable with this, let's quickly discuss this / sync. :blush: